### PR TITLE
Fix the configure test for pthread_getaffinity_np

### LINKED ISSUE #12411: FreeBSD pthread_getaffinity_np not detected
On FreeBSD 13.2, 64 bit x86 or 64 bit ARM, s.h does not define `HAS_BSD_GETAFFINITY_NP` as expected.  The function `pthread_getaffinity_np` is not found because command line to compile the test program needs -pthread or -lpthreads.

```
configure:18103: cc -o conftest    conftest.c -lm  >&5
ld: error: undefined symbol: pthread_getaffinity_np
>>> referenced by conftest.c
>>>               /tmp/conftest-2a7a19.o:(main)
cc: error: linker command failed with exit code 1 (use -v to see invocation)
```

### PR DESCRIPTION
**Pull request by shindere**
The C test program shouldinclude the right header files to avoid warnings.

### CODE REVIEW

File: configure.ac
Comment (omasanori) [inline configure.ac]:
```suggestion
  [AC_MSG_RESULT([GNU])
```
Purely cosmetic, though.

**Review by omasanori — COMMENTED** [review]
Looks fine overall to me, I have just a minor comment.

**Comment by shindere** [comment]
OK why not uppercasing GNU after all, it makes it coherent with BSD. Done.

**Review by omasanori — APPROVED** [review]

**Comment by shindere** [comment]
Masanori Ogino (2023/07/29 03:57 -0700):
> @omasanori approved this pull request.

Thanks :)

We'll have to wait for an approval by a core dev but thanks anyway.

File: configure.ac
Comment (dra27) [inline configure.ac:2154]: I think it's worth a comment that we permanently update `$CFLAGS` and `$CLIBS` so that subsequent tests can assume the presence of pthreads.

**Review by dra27 — APPROVED** [review]
A suggestion for an additional comment and I expect @shindere will want to tweak the history, but otherwise this looks fine to me.

**Comment by shindere** [comment]
David Allsopp (2023/08/04 02:54 -0700):
> @dra27 approved this pull request.

thanks.

> A suggestion for an additional comment and I expect @shindere will want to tweak the history, but otherwise this looks fine to me.
>
> >      CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
>      LIBS="$LIBS $PTHREAD_LIBS"
> -    AC_CHECK_FUNC([sigwait], [AC_DEFINE([HAS_SIGWAIT])])
> -    LIBS="$saved_LIBS"
> -    CFLAGS="$saved_CFLAGS"],
> +    AC_CHECK_FUNC([sigwait], [AC_DEFINE([HAS_SIGWAIT])])],
>
> I think it's worth a comment that we permanently update `$CFLAGS` and
> `$CLIBS` so that subsequent tests can assume the presence of pthreads.

Good idea, thnaks. History tweaked and comment added, finally.

**Pull request merged**
